src/gameobjects/units/cUnit.cpp

```diff
@@ -735,6 +735,36 @@ void cUnit::draw()
     const float scaledWidth = mapCamera->factorZoomLevel(bmp_width);
     const float scaledHeight = mapCamera->factorZoomLevel(bmp_height);
 
+    /*if (shadow) {
+        int destY = uy;
+
+        if (iType == CARRYALL) {
+            // adjust X and Y so it looks like a carry-all is 'higher up in the air'
+            destY = uy + 24; // TODO; do something with height here? the closer to target, the less distant the shadow?
+        }
+
+        SDL_Surface *stretchedShadow = SDL_CreateRGBSurface(0, scaledWidth, scaledHeight,32,0,0,0,255);
+        renderDrawer->FillWithColor(stretchedShadow, Color{255,0,255,255});
+        renderDrawer->maskedStretchBlit(shadow, stretchedShadow, 0, 0, bmp_width, bmp_height,
+                                        0, 0,
+                                        scaledWidth, scaledHeight);
+
+        renderDrawer->drawTransSprite(stretchedShadow, stretchedShadow, ux, destY);
+        SDL_FreeSurface(shadow);
+        SDL_FreeSurface(stretchedShadow);
+    }*/
+    Texture *shadow = cPlayer.getUnitShadowBitmap(iType);
+    if (shadow) {
+        cRectangle src = {start_x, start_y, bmp_width, bmp_height};
+        if (iType == CARRYALL) {
+            // start_x += 2;
+            // start_y += 2;
+            src = {start_x+2, start_y+2, bmp_width, bmp_height};
```

Comment:
so the translation between unit/body facing and which shadow 'piece' to draw happens here now, (instead in `getUnitShadowBitmap`). For now we can keep it there.

If this works btw, we can get rid of the commented out code above.

Reply:
I keep commented code to explain where this code comes from.

**long story:**
Previously, we would cut the sprites in the CPU memory and then send the piece to the display.
Currently, the entire image of all the sprites is in the GPU, and we ask the GPU to cut the sprite we're interested in.
This is faster and more convenient. Therefore, the old method returns the entire sprite, not a piece.